packages/core/admin/server/src/services/homepage.ts

```diff
@@ -1,129 +1,212 @@
-            fields,
+            sort: 'publishedAt:desc',
+            fields: meta.fields,
+            status: 'published',
           });
 
-          return documents.map((document) => {
-            /**
-             * Save the main field value before deleting it so we can use the common
-             * title key instead across all content types. Use the delete operator instead of
-             * destructuring or lodash omit for better type inference.
-             */
-            const mainFieldValue = document[mainField ?? 'documentId'];
-            delete document[mainField];
-
-            return {
-              data: {
-                ...document,
-                updatedAt: new Date(document.updatedAt),
-                title: mainFieldValue,
-              },
-              meta: {
-                model: contentTypeName,
-                kind: contentType.kind,
-                hasDraftAndPublish,
-              },
-            };
-          });
+          return formatDocuments(docs, meta);
         })
       );
 
       const overallRecentDocuments = recentDocuments
         .flat()
-        .sort((a, b) => b.data.updatedAt.valueOf() - a.data.updatedAt.valueOf())
+        .sort((a, b) => {
+          return b.data.publishedAt.valueOf() - a.data.publishedAt.valueOf();
+        })
         .slice(0, MAX_DOCUMENTS);
 
-      return Promise.all(
-        overallRecentDocuments.map(async (document) => {
-          /**
-           * Tries to query the other version of the document if draft and publish is enabled,
-           * so that we know when to give the "modified" status.
-           */
-          const { availableStatus } = await metadataService.getMetadata(
-            document.meta.model,
-            document.data,
-            {
-              availableStatus: document.meta.hasDraftAndPublish,
-              availableLocales: false,
-            }
-          );
-          const status = metadataService.getStatus(document.data, availableStatus);
-
-          return {
-            ...document.data,
-            ...document.meta,
-            status,
-          };
+      return addStatusToDocuments(overallRecentDocuments);
+    },
+    async getRecentlyUpdatedDocuments(): Promise<GetRecentDocuments.Response['data']> {
+      const allowedContentTypeNames = await getPermittedContentTypes();
+      // Fetch the configuration for each content type in a single query
+      const configurations = await getConfiguration(allowedContentTypeNames);
+      // Get the necessary metadata for the documents
+      const documentsMeta = await getDocumentsMetaData(allowedContentTypeNames, configurations);
+      // Now actually fetch and format the documents
+      const recentDocuments = await Promise.all(
+        documentsMeta.map(async (meta) => {
+          const docs = await strapi.documents(meta.uid).findMany({
+            limit: MAX_DOCUMENTS,
+            sort: 'updatedAt:desc',
+            fields: meta.fields,
```

Comment:
we could use an empty line in between

Reply:
Not I see where you mean? I think I have one after the findMany

Reply:
My comment is again in the wrong spot for some reason. I mean in between the two methods in the service

Reply:
oh right, yeah I'll do that